### PR TITLE
Dash move

### PR DESCRIPTION
**Pull request by bnmnetp**
Move db_dashboard.py to modules so it is not loaded on every request.

There are challenges with moving something like this to a module in that you need to use the `gluon.current` object to get at request, and db, etc.

The old unit tests are totally unable to cope with this craziness so I've disabled the two unit tests in test_dashboard.py (I've tested them extensively from the browser) and I'm going to write them in the style of test_server.py

@bjones1 my hand testing of these thing uncovered a problem with your utility function `get_course_row` by default it only returns a couple of columns from courses but anything that uses the `_sphinx_static_files.html` template will need much more.  I fixed the calls in dashboard.py to pass `db.courses.ALL` but there may be other problems lurking....

### CODE REVIEW

File: modules/db_dashboard.py [hunk 1]
```diff
@@ -2,11 +2,12 @@
...
 import six
+from gluon import current
 
-rslogger = logging.getLogger(settings.logger)
-rslogger.setLevel(settings.log_level)
+rslogger = logging.getLogger(current.settings.logger)
+rslogger.setLevel(current.settings.log_level)
 
-#db.define_table('dash_problem_answers',
+#current.db.define_table('dash_problem_answers',
```
Comment (bjones1) [inline modules/db_dashboard.py]: Why not just delete these? Are they used?

File: modules/db_dashboard.py [hunk 2]
```diff
@@ -17,7 +18,7 @@
 #)
 
 
-#db.define_table('dash_problem_user_metrics',
+#current.db.define_table('dash_problem_user_metrics',
```
Comment (bjones1) [inline modules/db_dashboard.py]: Same here.

File: modules/db_dashboard.py [hunk 3]
```diff
@@ -369,50 +372,51 @@ def load_chapter_metrics(self, chapter):
...
             rslogger.debug("ERROR - NO USER username={} course_id={}".format(username, self.course_id))
-            session.flash = 'Please make sure you are in the correct course'
-            redirect('/runestone/default/courses')
-
-        self.logs = db((db.useinfo.course_id==self.course.course_name) &
-                       (db.useinfo.sid == username) &
-                       (db.useinfo.timestamp >= self.course.term_start_date)).select(db.useinfo.timestamp,db.useinfo.sid, db.useinfo.event,db.useinfo.act,db.useinfo.div_id, orderby=~db.useinfo.timestamp)
-        self.db_chapter_progress = db((db.user_sub_chapter_progress.user_id == self.user.id)).select(db.user_sub_chapter_progress.chapter_id,db.user_sub_chapter_progress.sub_chapter_id,db.user_sub_chapter_progress.status)
+            current.session.flash = 'Please make sure you are in the correct course'
+            current.redirect('/runestone/default/courses')
```
Comment (bjones1) [inline modules/db_dashboard.py]: Use `URL('default', 'courses')` instead to avoid hard-coding the application.

File: modules/db_dashboard.py [hunk 4]
```diff
@@ -451,7 +455,7 @@ def load_assignment_metrics(self, username, studentView=False):
                                                "due_date":assign["duedate"].date().strftime("%m-%d-%Y")}
 
 # This whole object is a workaround because these strings
-# are not generated and stored in the db. This needs automating
+# are not generated and stored in the current.db. This needs automating
```
Comment (bjones1) [inline modules/db_dashboard.py]: A bit of overzealous find-and-replace...

File: tests/test_dashboard.py [hunk 5]
```diff
@@ -28,15 +28,21 @@ class TestDashboardEndpoints(unittest.TestCase):
     def setUp(self):
         global request, session, auth
         request = Request(globals()) # Use a clean Request object
+        request.folder = "/srv/web2py/applications/runestone/"
```
Comment (bjones1) [inline tests/test_dashboard.py]: Looks suspicious -- will only work in a Docker container...

**Comment by bjones1** [comment]
Hmmm, good point re: `get_course_row`. Hopefully, testing will uncover more cases...

File: tests/test_dashboard.py (hunk 5 above)
Comment (bnmnetp) [reply tests/test_dashboard.py]: Yes, but I'm going to rewrite it entirely so I don't think it really matters.

File: modules/db_dashboard.py (hunk 1 above)
Comment (bnmnetp) [reply modules/db_dashboard.py]: apparently not, since they've been commented out forever.

File: tests/test_dashboard.py (hunk 5 above)
Comment (bjones1) [reply tests/test_dashboard.py]: Sounds good.

File: modules/db_dashboard.py (hunk 1 above)
Comment (bnmnetp) [reply modules/db_dashboard.py]: I was actually on the verge of deleting them, but then I decided they they went along with the comment that followed.  That much of what we show in dashboard could be calculated and stored in one of the tables represented by the commented out definitions.

I suppose we could just copy them into an issue somewhere, but they aren't hurting anything where they are either.

Comment (bjones1) [reply modules/db_dashboard.py]: Sounds good.

File: modules/db_dashboard.py (hunk 4 above)
Comment (bnmnetp) [reply modules/db_dashboard.py]: This `IdConverter` object requires further investigation... I'm pretty sure its not used, and is insanely incomplete if it is, in fact thethings it is looking for are in the database.

**Pull request merged**